test/functional/change_stream.test.js

```diff
@@ -176,26 +171,25 @@ const pipeline = [
 ];
 
 describe('Change Streams', function () {
-  before(function () {
-    return setupDatabase(this.configuration, ['integration_tests']);
+  before(async function () {
+    return await setupDatabase(this.configuration, ['integration_tests']);
   });
 
-  beforeEach(function () {
+  beforeEach(async function () {
     const configuration = this.configuration;
     const client = configuration.newClient();
 
-    return client
-      .connect()
-      .then(() => {
-        const db = client.db('integration_tests');
-        return db.createCollection('test');
-      })
-      .then(
-        () => client.close(),
-        () => client.close()
-      );
+    await client.connect();
+    const db = client.db('integration_tests');
+    try {
+      await db.createCollection('test');
+    } catch {
+      // ns already exists, don't care
+    } finally {
+      await client.close();
+    }
   });
-  afterEach(() => mock.cleanup());
+  afterEach(async () => await mock.cleanup());
```

Comment:
Just wondering, does switching to async/await in the hooks fix something or is this just stylistic?

Reply:
Stylistic mostly but when debugging I can step over this line without going into the mocha lib to see the result of the cleanup.